src/DonationForms/Endpoints/FormActions.php

```diff
@@ -68,12 +69,19 @@ public function registerRoute()
     }
 
     /**
-     * @param  WP_REST_Request  $request
+     * @unreleased Add Nonce Verification
+     * @since      2.19.0
      *
-     * @return WP_REST_Response
+     * @param WP_REST_Request $request
+     *
+     * @return WP_REST_Response|WP_Error
      */
     public function handleRequest(WP_REST_Request $request)
     {
+        if ( ! wp_verify_nonce($request->get_header('X-WP-Nonce'), 'wp_rest')) {
```

Comment:
Does the REST Endpoint not already check for this nonce? I thought that was built into the WordPress system.

Reply:
Yeah. It's redundant. Resolved on e17092f9c6ffe64fc8fb598801f0a523810c264d